app/actions/websocket/index.ts

```diff
@@ -64,90 +68,126 @@ export async function handleClose(serverUrl: string, lastDisconnect: number) {
     });
 }
 
-function doFirstConnect(serverUrl: string) {
-    updateAllUsersSinceLastDisconnect(serverUrl);
-}
-
 async function doReconnect(serverUrl: string) {
-    const database = DatabaseManager.serverDatabases[serverUrl];
-    if (!database) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
         return;
     }
 
-    const system = await queryCommonSystemValues(database.database);
-    const lastDisconnectedAt = await queryWebSocketLastDisconnected(database.database);
-
-    // TODO consider fetch only and batch all the results.
-    await fetchMe(serverUrl);
-    await fetchMyPreferences(serverUrl);
-    const {config} = await fetchConfigAndLicense(serverUrl);
-    const {memberships: teamMemberships, error: teamMembershipsError} = await fetchMyTeams(serverUrl);
-    const {currentChannelId, currentUserId, currentTeamId, license} = system;
-    const currentTeamMembership = teamMemberships?.find((tm) => tm.team_id === currentTeamId && tm.delete_at === 0);
-
-    let channelMemberships: ChannelMembership[] | undefined;
-    if (currentTeamMembership) {
-        const {memberships, channels, error} = await fetchMyChannelsForTeam(serverUrl, currentTeamMembership.team_id, true, lastDisconnectedAt);
-        if (error) {
-            DeviceEventEmitter.emit(Events.TEAM_LOAD_ERROR, serverUrl, error);
-            return;
-        }
-        const currentUser = await queryCurrentUser(database.database);
-        const preferences = currentUser ? (await currentUser.preferences.fetch()) : [];
-        const teammateDisplayNameSetting = getTeammateNameDisplaySetting(preferences || [], system.config, license);
-        const directChannels = channels?.filter((c) => c.type === General.DM_CHANNEL || c.type === General.GM_CHANNEL);
+    const {database} = operator;
+    const tabletDevice = await isTablet();
+    const system = await queryCommonSystemValues(database);
+    const lastDisconnectedAt = await queryWebSocketLastDisconnected(database);
+
+    resetWebSocketLastDisconnected(operator);
+    let {config, license} = await fetchConfigAndLicense(serverUrl);
+    if (!config) {
+        config = system.config;
+    }
+
+    if (!license) {
+        license = system.license;
+    }
+
+    const fetchedData = await fetchAppEntryData(serverUrl, lastDisconnectedAt, system.currentTeamId);
+    const fetchedError = (fetchedData as AppEntryError).error;
+
+    if (fetchedError) {
+        return;
+    }
+
+    const {initialTeamId, teamData, chData, prefData, meData, removeTeamIds, removeChannelIds} = fetchedData as AppEntryData;
+    const rolesData = await fetchRoles(serverUrl, teamData.memberships, chData?.memberships, meData.user, true);
+
+    if (chData?.channels?.length) {
+        const teammateDisplayNameSetting = getTeammateNameDisplaySetting(prefData.preferences || [], config, license);
+        let directChannels: Channel[] = [];
```

Comment:
```suggestion
        let directChannels: Channel[];
```
There is no need to assign a value here. Also, on line 115 there is no need for the `?` since direcChannels will always be defined.